spec/controllers/link_check_rules_controller_spec.rb

```diff
@@ -5,13 +5,16 @@
   let(:link_check_rule) { build(:link_check_rule) }
   let(:user)            { create(:user, :admin) }
   let(:partner)         { build(:partner) }
+  let(:source)          { build(:source) }
 
   before(:each) do
+    allow_any_instance_of(Source).to receive(:update_apis)
     sign_in user
   end
 
   describe "GET 'index'" do
-    it 'should get all of the collection rules' do
+    it 'should get all of the valid collection rules' do
```

Comment:
should we add a test to cover a scenario:
`it should not show link rules that don't have a source`?

Reply:
good idea